frontend/src/stores/courses.js

```diff
@@ -10,9 +10,21 @@ const isAvailableByTime = (courseItem) =>
     dateFns.differenceInHours(courseItem.startDate, new Date()) > 1;
 
 const hasSufficientFund = (balance, courseItem) => courseItem.price <= balance;
-
-const isAvailable = (balance, courseItem) =>
-    hasSufficientFund(balance, courseItem) && isAvailableByTime(courseItem);
+const isAvailable = (balance, courseItem, authenticationStatus) => {
+    const satisfyTimeContrain = isAvailableByTime(courseItem);
+    const satisfyResourceConstrain = hasSufficientFund(balance, courseItem);
+    return {
+        isAvailable:
+            satisfyTimeContrain &&
+            satisfyResourceConstrain &&
+            authenticationStatus,
+        reasons: [
```

Comment:
Having an array of booleans or strings feel a little shady. What it should be from the perspective of end user is an array of `string` with errors present.